### DISCUSSION TITLE
How to select the MXSS parameter in Mt3dSsm

### DISCUSSION THREAD

**Discussion by Alvin-prc** [modflowpy/flopy#1765] [Q&A] [answered]
My model has 10 stress periods, and only one source sink item is ALLOWED in each stress period. Setting [MXSS] = 15, ERROR: ERROR: MAXIMUM NUMBER OF SINKS/SOURCES ALLOWED [MXSS] = 15 will be displayed:
INCREASE VALUE OF [MXSS] IN [SSM] PACKAGE INPUT FILE
Even if [MXSS] =100, it still won't work. If it is set to 200 or greater, it will work. I want to know why, even if [MXSS] = 15 is greater than 10, why can't the model work?
Even if I successfully run the MT3D model, the success return value is still False.
# SSM file 
###########################################################################
S1_x = 175.  # 污染源的x坐标
S1_y = 445.  # 污染源的y坐标
S1_row = int((Ly - S1_y)//delc)  # 污染源的行号，从0开始计数
S1_col = int(S1_x//delr)  # 污染源的列号，从0开始计数

itype = 15
mxss = 15  # Maximum number of sources and sinks
S1_flux = np.array([2073.6, 4838.4, 3715.2, 3024.0, 0., 0., 0., 0., 0., 0.])  # g/d
dtype = np.dtype([('k', '<i8'), ('i', '<i8'), ('j', '<i8'), ('css', '<f4'), ('itype', '<i8')])

ssm_data = {}
ssm_data[0] = [(0, S1_row, S1_col, S1_flux[0], itype)]

ssm = flopy.mt3d.Mt3dSsm(mt, mxss=mxss, stress_period_data=ssm_data, dtype=dtype)
# Write model input
mt.write_input()

# Try to delete the output files, to prevent use of older files
try:
    os.remove(os.path.join(workspace, 'MT3D001.UCN'))
except FileNotFoundError:
    pass

# Run the model
success, buff = mt.run_model()
if not success:
    raise Exception('MT3D did not terminate normally.')

**Comment by emorway-usgs** [comment, marked as answer]
Your flow model likely has more than 15 cells marked with some type of boundary condition.  If you're using constant head, specified inflow, or some other packages, the transport model needs to account for these when it is allocating space in memory.  Once you specified a value of 200 or greater, the transport model has enough memory allocated for what it detects in the linker file.  Perhaps you have some constant head cells specified in the .bas input file that you need to account for in the transport model?

**Reply by Alvin-prc** [reply]
Yes, my model has 80x80 cells, and the left and right edges are constant head boundaries and there are 160 cells that are set to constant head, but there is another problem, Even if I successfully run the MT3D model, the success return value is still False.
